queue_job/models/queue_job.py

```diff
@@ -506,3 +509,24 @@ def _test_job(self, failure_rate=0):
         _logger.info("Running test job.")
         if random.random() <= failure_rate:
             raise JobError("Job failed")
+
+    def _call_webhook(self, **kwargs):
+        webhook_url = kwargs.get("webhook_url", None)
```

Comment:
to make it even more useful out of the box, can you add a block

if kwargs.get("only_if_max_retries_reached", False):
   check for `job.retry >= job.max_retries`